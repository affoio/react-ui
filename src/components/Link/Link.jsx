import { bool, elementType } from 'prop-types';
import styled, { css } from 'styled-components';
import colors from '../../colors';

const propTypes = {
  component: elementType,
  disabled: bool,
};

const Link = styled.a`
  text-decoration: none;
  color: ${colors.indigoBlue[800]};
  cursor: pointer;
  display: inline-block;

  ${({ disabled }) =>
    disabled
      ? css`
          opacity: 0.5;
          cursor: text;
        `
      : css`
          &:hover,
          &:focus {
            text-decoration: underline;
          }
        `}
`;

Link.propTypes = propTypes;
Link.defaultProps = {
  component: 'a',
};

export default Link;
