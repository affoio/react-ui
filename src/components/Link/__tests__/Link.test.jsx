import React from 'react';
import { render, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Link from '../Link';

describe('<Link>', () => {
  it('renders correctly', () => {
    const tree = render(
      <Link href="/link">Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.</Link>
    );
    expect(tree).toMatchSnapshot();
  });

  it('render disabled Link', () => {
    const tree = renderer
      .create(<Link disabled>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.</Link>)
      .toJSON();

    expect(tree).toHaveStyleRule('opacity', '0.5');
  });

  it('render Link with another component', () => {
    const wrapper = mount(
      <Link as="span">Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.</Link>
    );

    expect(wrapper.find('span').length).toBe(1);
  });
});
