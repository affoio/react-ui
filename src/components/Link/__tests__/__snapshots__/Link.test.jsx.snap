// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link> renders correctly 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #4718b4;
  cursor: pointer;
  display: inline-block;
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<a
  class="c0"
  href="/link"
>
  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quaerat, voluptas.
</a>
`;
